fhir_zulip_nlp/fhir_zulip_nlp.py

```diff
@@ -155,80 +147,86 @@ def format_df(df: pd.DataFrame) -> pd.DataFrame:
     return df
 
 
-def query_categories(category_keywords: Dict[str, List[str]]) -> pd.DataFrame:
+def create_report1(
+    df: pd.DataFrame, category_keywords: Dict[str, List[str]] = KEYWORDS
+) -> (pd.DataFrame, pd.DataFrame):
+    """Report 1: counts and latest/oldest message timestamps"""
+    # No-results report
+    reports: List[Dict] = []
+    no_result_keywords: List[str] = []
+    for category, keywords in category_keywords.items():
+        for k in keywords:
+            df_kw = df[df['keyword'] == k]
+            df_kw = df_kw.sort_values(['timestamp'])  # oldest first
+            kw_report = {
+                'category': category,
+                'keyword': k,
+                # TODO: Check: If a message contains the keyword more than once, will it return more than 1 result? or
+                #  ...simply 1 message result? @Davera
+                'num_messages_with_keyword': len(df_kw),
+                'newest_message_datetime': format_timestamp(list(df_kw['timestamp'])[-1]) if len(df_kw) > 0 else None,
+                'oldest_message_datetime': format_timestamp(list(df_kw['timestamp'])[0]) if len(df_kw) > 0 else None,
+            }
+            if kw_report['num_messages_with_keyword'] == 0:
+                no_result_keywords.append(k)
+            reports.append(kw_report)
+
+    # Report 1
+    df_report1 = pd.DataFrame(reports)
+    df_report1 = format_df(df_report1)
+
+    # No results report
+    df_no_results = pd.DataFrame()
+    df_no_results['keywords_with_no_results'] = no_result_keywords
+
+    return df_report1, df_no_results
+
+
+def query_categories(category_keywords: Dict[str, List[str]] = KEYWORDS) -> pd.DataFrame:
     """Get all dictionaries"""
     # Load cache
     cache_df = pd.DataFrame()
-    cache_json_path = CONFIG['cache_outpath'].replace('.csv', '.json')
```

Comment:
- Delete: I decided to remove the caching of JSON. It wasn't really useful at all, and was just taking up lines and making everything more complicated / harder to read.